src/linear_solver_routines/block_solver_helpers/gentlemans_house.jl

```diff
@@ -0,0 +1,155 @@
+# This is file from RLinearAlgebra.jl
+"""
+    GentData{S, M<:Matrix{S}, V<:Vector{S}}
+
+This is a preallocated data structure for solving least squares problems with 
+Gentleman's algorithm. This data structure is not exported and thus is not 
+designed to be manipulated by the user.
+# Fields
+- `A::M`, The matrix to be solved.
+- `B::M`, The matrix that rows of `A` are brought to have QR decomposition
+applied.
+- `R::UpperTriangular`, The upper triangular part of `B`.
+- `tab::SubArray`, the last column of `B` where the constant vector entries corresponding
+to the rows of `A` that were brought to `B` are stored.
+- `v::V`, a buffer vector.
+- `bsize::Int64`, the number of rows transported to `B` at each iteration.
+
+Calling `Gent(A, bsize)` will allocate a `B` matrix with the `n + 1` columns and 
+`n + bsize + 1` rows where n is the number of columns of A. It also allocates a 
+buffer vector `v` with `n + 1` entries. Finally, it takes a view of the upper
+triangular part of `B[1:n, 1:n]` and a view of the last column of `B`.
```

Comment:
What does it mean for "it takes a view"?

Reply:
I think "allocates a view" is a more precise way of stating it.